File: Laba6/Laba6/Models/Task2Class.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Reflection.Emit;
using System.Text;
using System.Threading.Tasks;

namespace Laba6.Models
{
    class Bicycle
    {
        private string brand;
        private int year;
        private double price;
        private int gearCount;
        private bool isElectric;
        private string color;
        private double currentSpeed;

        public string Brand
        {
            get { return brand; }
            set { brand = value; }
        }

        public int Year
        {
            get { return year; }
            set { year = value; }
        }

        public double Price
        {
            get { return price; }
            set { price = value; }
        }

        public int GearCount
        {
            get { return gearCount; }
            set { gearCount = value; }
        }

        public bool IsElectric
        {
            get { return isElectric; }
            set { isElectric = value; }
        }

        public string Color
        {
            get { return color; }
            set { color = value; }
        }

        public double CurrentSpeed
        {
            get { return currentSpeed; }
            set { currentSpeed = value; }
        }


        public Bicycle()
        {
            brand = "Unknown";
            year = 2022;
            price = 0.0;
            gearCount = 1;
            isElectric = false;
            color = "Black";
            currentSpeed = 0.0;
        }

        public Bicycle(string brand, int year, double price, int gearCount, bool isElectric, string color, double currentSpeed)
        {
            this.brand = brand;
            this.year = year;
            this.price = price;
            this.gearCount = gearCount;
            this.isElectric = isElectric;
            this.color = color;
            this.currentSpeed = currentSpeed;
        }


        public void Accelerate(double speedIncrease)
        {
            currentSpeed += speedIncrease;
            Console.WriteLine($"Accelerating to {currentSpeed} km/h.");
        }

        public void Brake()
        {
            currentSpeed = 0.0;
            Console.WriteLine("Bicycle stopped.");
        }

        public string DisplayInfo()
        {
            return 
             $"Brand: {brand}"
             + $" Year: {year}"
             + $" Price: ${price}\n"
             + $" Gears: {gearCount}"
             + $" Electric: {isElectric}"
             + $" Color: {color}\n"
             + $" Current Speed: {currentSpeed} km/h";
        }
    }
}

File: Laba6-Python/TaskModels/task2Class.py
class Bicycle:
    def __init__(self, brand="Unknown", year=2022, price=0.0, gear_count=1, is_electric=False, color="Black", current_speed=0.0):
        self.brand = brand
        self.year = year
        self.price = price
        self.gear_count = gear_count
        self.is_electric = is_electric
        self.color = color
        self.current_speed = current_speed

    def accelerate(self, speed_increase):
        self.current_speed += speed_increase
        print(f"Accelerating to {self.current_speed} km/h.")

    def brake(self):
        self.current_speed = 0.0
        print("Bicycle stopped.")

    def display_info(self):
        return (
            f"Brand: {self.brand} "
            f"Year: {self.year} "
            f"Price: ${self.price}\n"
            f"Gears: {self.gear_count} "
            f"Electric: {self.is_electric} "
            f"Color: {self.color}\n"
            f"Current Speed: {self.current_speed} km/h"
        )